roles/common/tasks/networking.yml

```diff
@@ -13,6 +13,36 @@
   with_items: network_interfaces_d
   tags: ['prereboot']
 
+- name: set default network gateway
+  set_fact: dfl_nw_gw="{{ ansible_default_ipv4.network.split('.')[0] }}.{{ ansible_default_ipv4.network.split('.')[1] }}.{{ansible_default_ipv4.network.split('.')[2] }}.{{ (ansible_default_ipv4.network.split('.')[3] | int) + 1 }}"
+  when: update_routes|default('false')|bool
+
+- name: add static route for multi subnet
+  command: "/sbin/ip route replace {{ item }} via {{ dfl_nw_gw }}"
+  when: (not (ansible_default_ipv4.network ==  ( item | ipaddr('network') ))) and (update_routes|default('false')|bool)
+  with_items: "{{ undercloud_cidr }}"
+
+- name: Check if interface has its own config file
+  stat: path=/etc/network/interfaces.d/{{ ansible_default_ipv4.interface }}.cfg
+  register: found_in_interfaces_d 
+
+- name: Set network config file
+  set_fact: nw_config_file="/etc/network/interfaces.d/{{ ansible_default_ipv4.interface }}.cfg"
+  when: found_in_interfaces_d.stat.exists is defined and found_in_interfaces_d.stat.exists
+
+- name: Set network config file
+  set_fact: nw_config_file="/etc/network/interfaces"
+  when: not found_in_interfaces_d.stat.exists|bool
+
+- name: Check for static routes in {{ nw_config_file }}
```

Comment:
+  register: routes_defined
+  failed_when: routes_defined.rc != 0

To fail that node if grep fails. Otherwise grep command itself will not fail irrespective of whether the line exists or not.